Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/local/source/LocalStatisticsRetrievalScheduler.kt

```diff
@@ -0,0 +1,40 @@
+package de.rki.coronawarnapp.statistics.local.source
+
+import de.rki.coronawarnapp.statistics.local.FederalStateToPackageId
+import de.rki.coronawarnapp.statistics.local.storage.LocalStatisticsConfigStorage
+import de.rki.coronawarnapp.util.device.ForegroundState
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class LocalStatisticsRetrievalScheduler @Inject constructor(
+    foregroundState: ForegroundState,
+    localStatisticsConfigStorage: LocalStatisticsConfigStorage,
+) {
+    private val lastActiveStates = mutableSetOf<FederalStateToPackageId>()
+
+    val updateStatsTrigger = combine(
+        foregroundState.isInForeground,
+        localStatisticsConfigStorage.activeStates.flow
+    ) { isInForeground, statistics ->
+        val hasNewStats = statistics.any {
+            val isNew = !lastActiveStates.contains(it)
+            if (isNew) lastActiveStates.add(it)
+            isNew
+        }
```

Comment:
What happens if the user switches to `State A`, then to `State B` and then back to `State A`, does it still refresh data for `State A` again?

Reply:
probably not, i've replaced the logic with a contains all check